core/txpool/tx_pool.go

```diff
@@ -222,7 +153,7 @@ func (config *TxPoolConfig) sanitize() TxPoolConfig {
 // current state) and future transactions. Transactions move between those
 // two states over time as they are received and processed.
 type TxPool struct {
```

Comment:
Maybe (and this might not work at all), we could make the interface name `TxPool`, and have this type be `defaultPool` or `standardTxPool`, or just `txPool`. 

Regardless -- since you've made an interface, does this type need to be public?

Reply:
Yes we can do it and also privatize the standardTxPool. Only the constructor needs to be public